toolkit/imageconfigs/additionalconfigs/coal.sh

```diff
@@ -24,13 +25,60 @@ echo "/data/containerd /var/lib/containerd none bind 0 0" >> /etc/fstab
 # Enable initrd to break into a shell
 #sed -i "s/rd.shell=0 rd.emergency=reboot/rd.shell=1 rd.break=pre-pivot/" /boot/grub2/grub.cfg
 
-# Ensure overlay driver is available in initrd
-echo "add_drivers+=\" overlay \"" >> /etc/dracut.conf.d/01-coal.conf
-# Enable systemd-repart in the initrd
-echo "add_dracutmodules+=\" systemd-repart \"" >> /etc/dracut.conf.d/01-coal.conf
+# UKI logic
+EFIDIR="BOOT"
+KERNEL_VERSION=""
 
 mkdir -p /etc/repart.d
 echo -e "[Partition]\nType=esp" > /etc/repart.d/10-coal.conf; echo -e "[Partition]\nType=linux-generic" > /etc/repart.d/11-coal.conf; echo -e "[Partition]\nType=linux-generic" > /etc/repart.d/12-coal.conf; echo -e "[Partition]\nType=linux-generic" > /etc/repart.d/13-coal.conf
 
+# Image generation is done in a chroot environment, so running `uname -r`
+# will return the version of the host running kernel. This function works
+# under the assumption that exactly one kernel is installed in the end image.
+get_kernel_version() {
+    kernel_modules_dir="/usr/lib/modules"
+    KERNEL_VERSION="$(ls $kernel_modules_dir)"
+}
+
+# symlink /boot/efi to ../efi
+cp -a /boot/efi/. /efi
+rm -rf /boot/efi
+ln -s ../efi /boot/efi
+
+# The shim has its default boot-loader filename built in as grubx64.efi.
+# To switch to systemd-boot, we overwrite that file location with the
+# sd-boot EFI binary as a workaround.
+cp /lib/systemd/boot/efi/systemd-bootx64.efi /efi/EFI/$EFIDIR/grubx64.efi
+
+echo "fstab-before:"
+cat /etc/fstab
+
+# empty /etc/fstab file
+# sed -i '/\/ ext4/d' /etc/fstab
+# sed -i '/\/efi vfat/d' /etc/fstab
+echo > /etc/fstab
+
+# echo "fstab-after:"
+# cat /etc/fstab
+
+# copy UKI into the ESP
+mkdir -p /efi/EFI/Linux
+get_kernel_version
```

Comment:
This function is only called once and is implicitly setting an env var for its caller. Just call `KERNEL_VERSION="$(ls /usr/lib/modules)"` directly here.

Reply:
Fixed in "round 2" of addressing comments!